CourierAppBackend/src/main/java/com/fdmgroup/courierapp/controller/CustomerController.java

```diff
@@ -44,4 +58,43 @@ public ResponseEntity<ResponseOrderHistory> getCustomerOrderHistory(@RequestHead
         ResponseOrderHistory responseOrderHistory = new ResponseOrderHistory("Success", "Fetch success", orderDetailsList);
         return new ResponseEntity<>(responseOrderHistory, HttpStatus.OK);
     }
+
+    @PutMapping("/{orderId}")
+    public ResponseEntity<ResponseOrderUpdate> updateCustomerOrder(
+            @PathVariable("orderId") Long orderId,
+            @RequestHeader("username") String username,
+            @RequestBody RequestOrderUpdate requestOrderUpdate
+    ) {
+        CustomerOrder customerOrder;
+        Customer customer;
+        try {
+            customerOrder = customerOrderService.findByCustomerOrderId(orderId);
+            customer = customerService.findByUsername(username);
+        } catch (OrderNotFoundException | CustomerNotFoundException e) {
+            return new ResponseEntity<>(new ResponseOrderUpdate("Failed", e.getMessage()), HttpStatus.OK);
+        }
+
+        int lastIndex = customerOrder.getStatus().size() - 1;
+
+        if (!customerOrder.getStatus().get(lastIndex).getStatus().equals(StatusEnum.ORDER_CREATED)) {
```

Comment:
might need to change List<Status> to Stack<Status>

Reply:
yeah sounds good idea to change to stack